plugins/ingest/src/test/java/org/elasticsearch/ingest/IngestDocumentTests.java

```diff
@@ -621,6 +619,34 @@ public void testRemoveEmptyField() {
         }
     }
 
+    public void testAccessSourceViaTemplate() {
+        TemplateService templateService = TestTemplateService.instance();
+
+        assertThat(ingestDocument.hasField("index"), is(false));
+        ingestDocument.setFieldValue(templateService.compile("{{_index}}"), ValueSource.wrap("{{_index}}", templateService));
+        assertThat(ingestDocument.getFieldValue("index", String.class), equalTo("index"));
+        ingestDocument.removeField(templateService.compile("{{_index}}"));
+        assertThat(ingestDocument.hasField("index"), is(false));
```

Comment:
oh boy, so this test sets a field called "index" to value "index" which is contained in the "_index" field. I must admit I would try to make it a bit easier to read using different values if possible.